code/modules/admin/secrets.dm

```diff
@@ -1,115 +1,66 @@
-			<A href='?src=[REF(src)];[HrefToken()];secrets=eagles'>Egalitarian Station Mode (All doors are open access)</A><BR>
-			<A href='?src=[REF(src)];[HrefToken()];secrets=ancap'>Anarcho-Capitalist Station Mode (More things cost money)</A><BR>
-			<A href='?src=[REF(src)];[HrefToken()];secrets=blackout'>Break all lights</A><BR>
-			<A href='?src=[REF(src)];[HrefToken()];secrets=whiteout'>Fix all lights</A><BR>
-			<A href='?src=[REF(src)];[HrefToken()];secrets=floorlava'>The floor is lava! (DANGEROUS: extremely lame)</A><BR>
-			<BR>
-			<A href='?src=[REF(src)];[HrefToken()];secrets=flipmovement'>Flip client movement directions</A><BR>
-			<A href='?src=[REF(src)];[HrefToken()];secrets=randommovement'>Randomize client movement directions</A><BR>
-			<A href='?src=[REF(src)];[HrefToken()];secrets=custommovement'>Set each movement direction manually</A><BR>
-			<A href='?src=[REF(src)];[HrefToken()];secrets=resetmovement'>Reset movement directions to default</A><BR>
-			<BR>
-			<A href='?src=[REF(src)];[HrefToken()];secrets=changebombcap'>Change bomb cap</A><BR>
-			<A href='?src=[REF(src)];[HrefToken()];secrets=masspurrbation'>Mass Purrbation</A><BR>
-			<A href='?src=[REF(src)];[HrefToken()];secrets=massremovepurrbation'>Mass Remove Purrbation</A><BR>
-			"}
-
-	dat += "<BR>"
-
-	if(check_rights(R_DEBUG,0))
-		dat += {"
-			<B>Security Level Elevated</B><BR>
-			<BR>
-			<A href='?src=[REF(src)];[HrefToken()];secrets=maint_access_engiebrig'>Change all maintenance doors to engie/brig access only</A><BR>
-			<A href='?src=[REF(src)];[HrefToken()];secrets=maint_access_brig'>Change all maintenance doors to brig access only</A><BR>
-			<A href='?src=[REF(src)];[HrefToken()];secrets=infinite_sec'>Remove cap on security officers</A><BR>
-			<BR>
-			"}
-
-	dat += "</BODY></HTML>"
-
-	usr << browse(dat.Join(), "window=secrets")
+	new /datum/tgui_secrets_panel(usr)
 	return
 
-
-
-
-
-/datum/admins/proc/Secrets_topic(item,href_list)
+/datum/tgui_secrets_panel
+	var/client/holder //client of whoever is using this datum
+
+/datum/tgui_secrets_panel/ui_state(mob/user)
+	return GLOB.admin_state
+
+/datum/tgui_secrets_panel/ui_interact(mob/user, datum/tgui/ui)
+	ui = SStgui.try_update_ui(user, src, ui)
+	if(!ui)
+		// Open UI
+		ui = new(user, src, "SecretsPanel")
+		ui.open()
+
+/datum/tgui_secrets_panel/New(user) //user can either be a client or a mob
+	if (user) //Prevents runtimes on datums being made without clients
+		setup(user)
+
+/datum/tgui_secrets_panel/proc/setup(user) //H can either be a client or a mob
+	if (istype(user,/client))
+		var/client/user_client = user
+		holder = user_client //if its a client, assign it to holder
+	else
+		var/mob/user_mob = user
+		holder = user_mob.client //if its a mob, assign the mob's client to holder
+	ui_interact(holder.mob)
+
+/datum/tgui_secrets_panel/ui_data(mob/user)
+	var/list/data = list()
+	var/client/rights = holder
```

Comment:
Do not give bibby full access to the panel

Reply:
fixed